server/app/models/MaterialModel.ts

```diff
@@ -0,0 +1,17 @@
+import { Document, Schema } from 'mongoose';
+
+export interface IMaterial extends Document {
+  description: string;
+  name: string;
+  stock: number;
+  weight: string;
+  price: number;
+}
+
+export const MaterialSchema: Schema = new Schema({
+  description: { type: Schema.Types.String },
+  name: { type: Schema.Types.String, required: true, unique: true, immutable: true },
```

Comment:
Can a key be mutable? why do we need to add "immutable: true"?

Reply:
unique in mongo does not mean you cant update, it only specifies that a name can only appear once in a collection. By default you can mutate even if its unique. I set it to immutable for now since were treating name like a key and we should mutate it

Reply:
Thank you, Scrum Master!! 🙏That makes so much sense!